tests/functional/z_cluster/test_scale_mon_pods.py

```diff
@@ -0,0 +1,324 @@
+            assert self.storagecluster_obj.patch(
+                params=params,
+                format_type="merge",
+            ), log.error("Mon count should not be updated value other than 3 and 5")
+        if self.are_rack_labels_present():
+            self.nodes = get_worker_nodes()
+            self.racks = [
+                "rack{}".format(i % (len(self.nodes) // 2))
+                for i in range(len(self.nodes))
+            ]
+
+            self.assign_dummy_racks()
+
+    @post_ocs_upgrade
+    @pytest.mark.polarion_id("OCS-5664")
+    @pytest.mark.polarion_id("OCS-5665")
+    @pytest.mark.polarion_id("OCS-5667")
+    @pytest.mark.polarion_id("OCS-5668")
+    def test_scale_mons_in_cluster_to_five(self, threading_lock):
+        """
+
+        A Testcase to add five mon pods to the cluster when the failure domain value is greater than five
+
+        This test looks if failure domain is greater than five, if yes it will update the monCount to five
+        and will wait for the CephMonLowNumber alert to get cleared
+
+        """
+
+        target_msg = "The current number of Ceph monitors can be increased in order to improve cluster resilience."
+        target_label = constants.ALERT_CEPHMONLOWCOUNT
+
+        list_mons = self.ceph_cluster.get_mons_from_cluster()
+        assert len(list_mons) < self.mon_count, pytest.skip(
+            "INVALID: Mon count is already above three."
+        )
+
+        # Sleep for 2 minutes so that alert can be generated
+        time.sleep(120)
+
+        api = prometheus.PrometheusAPI(threading_lock=threading_lock)
+        alerts_response = api.get(
+            "alerts", payload={"silenced": False, "inhibited": False}
+        )
+        if not alerts_response.ok:
+            log.error(f"got bad response from Prometheus: {alerts_response.text}")
+        prometheus_alerts = alerts_response.json()["data"]["alerts"]
+
+        log.info("verifying that alert is generated to update monCount to five")
+        try:
+            prometheus.check_alert_list(
+                label=target_label,
+                msg=target_msg,
+                alerts=prometheus_alerts,
+                states=["firing"],
+                severity="info",
+                ignore_more_occurences=True,
+            )
+            test_pass = True
+        except AssertionError:
+            pytest.fail(
+                "Failed to get CephMonLowCount warning when failure domain is updated to five"
+            )
+
+        if test_pass:
+            params_neg = '{"spec":{"managedResources":{"cephCluster":{"monCount": 4}}}}'
+            params = '{"spec":{"managedResources":{"cephCluster":{"monCount": 5}}}}'
+            try:
+                assert self.storagecluster_obj.patch(
+                    params=params_neg,
+                    format_type="merge",
+                ), log.error("Mon count should not be updated value other than 3 and 5")
```

Comment:
Correct me if I am wrong, but I think this should be `assert not self.storagecluster_obj.patch(` since we expect the patch to return `False`. Or should we perform a different check here?